### PR TITLE
Add "meta" support to hapi decorator

### PR DESCRIPTION
**Pull request by nickrobillard**
Here is a PR for #88

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -128,6 +137,31 @@ describe('Calibrate', () => {
         expect(res.payload).to.equal(validOutput);
     });
 
+    it('response using hapi decorator includes meta', async () => {
+
+        const server = new Hapi.Server();
+
+        await server.register({
+            plugin: Calibrate.hapi,
+            options: { onResponse: false }
```
Comment (johnbrett) [inline test/index.js]: why is `onResponse` needed here? I think this might be non-obvious to users, and we may need to specify this in docs if required

**Comment by johnbrett** [comment]
@nickrobillard just one question, but other than all is looking solid on this!

File: test/index.js (hunk 1 above)
Comment (nickrobillard) [reply test/index.js]: Yeah I realize now that I didn't consider the case of `onResponse: true`. This case is a bit strange. To support meta here means that the return format would need to be something special like: `{data, meta}` or even an array like `[data, meta]`, which would obviously be a breaking change. And it doesn't quite feel right. I'm not sure how to proceed here. Part of me says that the current decorator functionality - disabling half of calibrate's feature set by not allowing meta to be set - is not right. But adding meta support when `onResponse: true` means that the return format is now "special" and must follow a specific format, which may defeat the purpose of this option. (I would also guess that anyone who uses `onResponse: true` does not care about meta.) I'm curious what your thoughts are - I may be missing something.

Comment (johnbrett) [reply test/index.js]: I don't see `onResponse` anywhere in the API docs, can you link me to an explanation of that it does ?

Comment (nickrobillard) [reply test/index.js]: I also don't see `onResponse` documented anywhere. That being said, its purpose seems pretty straight forward. When it is `true`, it automatically wraps all route handler responses with calibrate. I can only assume it was added as a way to consistently wrap every response without having to change your route handler code. 

```
register(server, { onResponse = true }) {

        if (onResponse) {
            const preResponse = function (request, h) {

                const data = request.response.isBoom ? request.response : request.response.source;
                return internals.calibrate(data);
            };

            server.ext('onPreResponse', preResponse);

            return;
        }
```
With this in mind, hopefully my above comment is more understandable.

Comment (johnbrett) [reply test/index.js]: Yes, this all makes more sense now, thanks for explaining. I actually think this `onResponse` will cause issues, especially being on by default and needs further thought. 

This should be disabled by default and opt in only, but to be done in a separate PR and will be a breaking change to `calibrate` - just an FYI.

Other than that this change looks good, I will merge approve and merge, nice work @nickrobillard! I've also given you maintainer access to the repo!

**Review by johnbrett — APPROVED** [review]

**Comment by nickrobillard** [comment]
Great, thank you for adding this!

File: test/index.js (hunk 1 above)
Comment (nickrobillard) [reply test/index.js:146]: That makes sense - and this resolves my original issue. I'm not concerned with the `onResponse: true` case supporting meta (in my mind it shouldn't). Thanks for the quick action here!

**Pull request merged**